src/refactor/interactiveHandler.py

```diff
@@ -0,0 +1,112 @@
+
+from system import services
+from inputParser import inputParser
+from receiveParser import receiveParser
+from embedCSP import embedPacket
+import time
+
+class interactiveHandler:
+    def __init__(self):
+        self.services = services
+        self.inParser = inputParser()
+
+        # TODO: This is bad, make it good when fixing inputParser
+        self.appIdx = 0
+        self.serviceIdx = 2
+        self.subserviceIdx = 4
+
+    def getTransactionObject(self, command : str, networkHandler):
+        transactObj = None
+        # TODO: Make this less bad after fixing inputParser badness
+        tokens = self.inParser.lexer(command)
+        if tokens[self.serviceIdx] == "HOUSEKEEPING" and tokens[self.subserviceIdx] == "GET_HK":
+            transactObj = getHKTransaction(command, networkHandler)
+        elif tokens[self.serviceIdx] == "CLI":
+            transactObj = satcliTransaction(command, networkHandler)
+        elif tokens[self.serviceIdx] == "SCHEDULER" and (tokens[self.subserviceIdx] in ['SET_SCHEDULE', 'DELETE_SCHEDULE', 'REPLACE_SCHEDULE']):
+            transactObj = schedulerTransaction(command, networkHandler)
+        elif tokens[self.serviceIdx] == "TIME_MANAGEMENT" and tokens[self.subserviceIdx] == "SET_TIME":
+            transactObj = setTimeTransaction(command, networkHandler)
+        else:
+            transactObj = baseTransaction(command, networkHandler)
+
+        return transactObj
+    
+class baseTransaction:
+    def __init__(self, command, networkHandler):
+        self.networkHandler = networkHandler
+        self.inputParse = inputParser()
+        self.returnParse = receiveParser()
+        self.command = command
+        self.pkt = self.inputParse.parseInput(self.command)
+        self.dst = self.pkt["dst"]
+        self.dport = self.pkt['dport']
+        self.args = self.pkt['args']
+
+    def send(self):
+        self.networkHandler.send(self.dst, self.dport, self.args)
+
+    def receive(self):
+        return self.networkHandler.receive(self.dst, self.dport, 10000)
+        
+    def parseReturnValue(self, data):
+        return self.returnParse.parseReturnValue(self.dport, data, len(data))
+
+    def execute(self):
+        self.send()
+        ret = self.receive()
+        return self.parseReturnValue(ret)
+
+class setTimeTransaction(baseTransaction):
+    def execute(self):
+        tokens = self.inputParse.lexer(self.command)
+        time_param = tokens[-2]
+        now = (int(time_param))
+        if (now == 0):
+            now = int(time.time())
+
+        tokens[-2] = str(now)
+        self.pkt = self.inputParse.parseInput("".join(tokens))
+        self.args = self.pkt['args']
+        self.send()
+        return self.parseReturnValue(self.receive())
+
+
+class schedulerTransaction(baseTransaction):
+    def execute(self):
+        tokens = self.inputParse.lexer(self.command)
+        file_param = tokens[-2]
```

Comment:
Is the schedule file now an argument to the command? [ie. ex2.scheduler.set_schedule(flightschedules/schedule1.txt)]

Reply:
yes